src/authAgents/SessionAuthAgent.js

```diff
@@ -55,8 +61,14 @@ define(["structures/CAPIError", "storages/LocalStorage"], function (CAPIError, L
              */
             this._storage = storage || new LocalStorage();
 
-            if ( credentials ) {
-                this.setCredentials(credentials);
+            if ( authInfo ) {
+                if ( authInfo.login && authInfo.password ) {
+                    this.setCredentials(authInfo);
+                } else if ( authInfo.csrfToken && authInfo.identifier && authInfo.name && authInfo.href ) {
+                    this._storeSessionInfo(authInfo);
+                } else {
+                    throw new CAPIError("Invalid authInfo parameter");
```

Comment:
IMHO that's misleading. The `authInfo` can contain others properties than the ones checked here without issue.